content/docs/devops-tips/scaling-cert-manager.md

```diff
@@ -0,0 +1,134 @@
+> a proposal for a cert-manager configuration option to disable client-side rate-limiting.
+>
+> 🔗 Read [`kubernetes#111880`: Disable client-side rate-limiting when AP&F is enabled](https://github.com/kubernetes/kubernetes/issues/111880);
+> a proposal that the `kubernetes.io/client-go` module should automatically use server-side rate-limiting when it is enabled.
+>
+> 🔗 Read about other projects that disable client-side rate limiting: [Flux](https://github.com/fluxcd/pkg/issues/269).
+>
+> 📖 Read [API documentation for ControllerConfiguration](../reference/api-docs.md#controller.config.cert-manager.io/v1alpha1.ControllerConfiguration) for a description of the `kubernetesAPIQPS` and `kubernetesAPIBurst` configuration options.
+
+## Restrict the use of large RSA keys
+
+Certificates with large RSA keys cause cert-manager to use more CPU resources.
+When there are insufficient CPU resources, the reconcile queue length grows,
+which delays the reconciliation of all Certificates.
+A user who has permission to create a large number of RSA 4096 certificates,
+might accidentally or maliciously cause a denial of service for other users on the cluster.
+
+> 📖 Learn [how to enforce an Approval Policy](../policy/approval/README.md), to prevent the use of large RSA keys.
+>
+> 📖 Learn [how to set Certificate defaults automatically](../tutorials/certificate-defaults/README.md), using tools like Kyverno.
+
+
+## Set `revisionHistoryLimit: 1` on all Certificate resources
+
+By default, cert-manager will keep all the CertificateRequest resources that **it** creates
+([`revisionHistoryLimit`](../reference/api-docs.md#cert-manager.io/v1.CertificateSpec)):
+
+> The maximum number of CertificateRequest revisions that are maintained in
+> the Certificate's history. Each revision represents a single
+> `CertificateRequest` created by this Certificate, either when it was
+> created, renewed, or Spec was changed. Revisions will be removed by oldest
+> first if the number of revisions exceeds this number.
+>  If set, `revisionHistoryLimit` must be a value of `1` or greater. If unset
+> (`nil`), revisions will not be garbage collected. Default value is `nil`.
+
+On a busy cluster these will eventually overwhelm your Kubernetes API server;
+because of the memory and CPU required to cache them all and the storage required to save them.
+
+Use a tool like Kyverno to override the `Certificate.spec.revisionHistoryLimit` for all namespaces.
+
+> 📖 Adapt [the Kyverno policies in the tutorial: how to set Certificate defaults automatically](../tutorials/certificate-defaults/README.md),
+> to override rather than default the `revisionHistoryLimit` field.
+>
+> 📖 Learn [how to set `revisionHistoryLimit` when using Annotated Ingress resources](../usage/ingress.md#supported-annotations).
+>
+> 🔗 Read [`cert-manager#3773`: Certificate revision history limit](https://github.com/cert-manager/cert-manager/pull/3773),
```

Comment:
@wallrj It is quite hard to find a why in that pull request, didn't we discuss in one of our standups that we should set a sane default value?